components/layout/LayoutSiteMessageModal.vue

```diff
@@ -0,0 +1,139 @@
+<template>
+  <UModal
+    v-model:open="showModel"
+    class="bg-sta-gray max-w-3/4 w-fit"
+    :close="false"
+    :dismissible="!preventDismiss"
+    @after:leave="dismissBanner"
+  >
+    <template #description>
+      <div class="overflow-scroll max-h-[80vh]">
+        <div class="flex items-center justify-between mb-2">
+          <h1 class="text-xl text-white font-bold">
+            {{ siteMessage.title }}
+          </h1>
+          <UiStaButton
+            id="maintenanceBannerDismiss"
+            class="text-xl -my-2 text-sta-orange hover:text-sta-orange-dark"
+            icon="circle-xmark"
+            :disabled="preventDismiss"
+            @click="dismissBanner"
+          />
+        </div>
+        <div class="text-white text-lg">
+          <div v-for="(line, index) in formattedMessage" :key="index">
+            <p v-if="line.trim() !== ''">{{ line }}</p>
+            <br v-else />
+          </div>
+        </div>
+      </div>
+    </template>
+  </UModal>
+</template>
+
+<script>
+import cookie from 'js-cookie';
+import { UpcomingSiteMessagesDocument } from '~/graphql/codegen/operations';
+import humanizeDuration from 'humanize-duration';
+import { DateTime } from 'luxon';
+
+export default {
+  name: 'LayoutSiteMessageModal',
+  props: {
+    location: {
+      required: true,
+      type: String
+    }
+  },
+  data() {
+    return {
+      preventDismiss: false,
+      siteMessage: null,
+      dismissedIds: [],
+      showModel: false
+    };
+  },
+  computed: {
+    isOngoing() {
+      return this.siteMessage.eventStart < DateTime.now().toISO();
+    },
+    formattedMessage() {
+      // Split around new lines and trim each line
+      // Allows us to show multi-line messages without needing HTML in the message
+      // (because that's bad)
+      const msg = this.siteMessage?.message || '';
+      return msg.split('\n').map((line) => line.trim());
+    }
+  },
+  mounted() {
+    // Need to store the alert's id in the cookie to check if it's been superceded
+    this.dismissedIds = cookie.get('siteMessageModalDismissed')
+      ? cookie.get('siteMessageModalDismissed').split(',')
```

Comment:
According to GraphQL, the Maintainence modal has been dismissed (and, in fact, I did dismiss it when I first loaded the page). However, upon then clicking 'refresh', the siteMessage displayed again, even though you can see it is dismissed according to the cookie too.

<img width="544" height="356" alt="image" src="https://github.com/user-attachments/assets/8e266747-c1bb-42d9-825c-f54b470fbce7" />

<img width="1901" height="550" alt="image" src="https://github.com/user-attachments/assets/50256db5-1eb0-44bc-a850-5a802c0c315e" />